Sources/AWSLambdaRuntime/LambdaRuntimeClient.swift

```diff
@@ -363,7 +363,19 @@ final actor LambdaRuntimeClient: LambdaRuntimeClientProtocol {
             )
             channel.closeFuture.whenComplete { result in
                 self.assumeIsolated { runtimeClient in
+
+                    // resume any pending continuation on the handler
+                    if case .connected(_, let handler) = runtimeClient.connectionState {
+                        if case .connected(_, let lambdaState) = handler.state {
```

Comment:
This shouldn't be necessary, as `LambdaChannelHandler.channelInactive` will have resumed the continuation (with error `ioOnClosedChannel`). None of the tests trigger this code either.

Reply:
`LambdaChannelHandlerHandler.channelInnactive` is indeed called when the server closes a connection but never in the state `waitingForNextInvocation`.

When the server closes the connection, the state in the `LambdaChannelHandler` is 
`connected` and `LambdaState.idle` 

To reproduce what I observe:

1. `while true; do MAX_INVOCATIONS=1 MODE=json PORT=7777 LOG_LEVEL=trace swift run MockServer; sleep 1; done`
2. from another terminal: `cd Examples/HelloJSON ; LAMBDA_USE_LOCAL_DEPS=../.. LOG_LEVEL=trace AWS_LAMBDA_RUNTIME_API=127.0.0.1:7777 swift run`

You'll that when the server closes the connection, `LambdaChannelHandler.channelInactive()` is called with `self state == .connected(_, .idle)`

Reply:
Done, https://github.com/swift-server/swift-aws-lambda-runtime/pull/545/commits/b0234f5c3f576464c52caf9c1fe12702cc8e6a9b

Reply:
As you noticed, the `connnectionToControlPlaneLost` error was not triggered by the previous commit.  This commit adds a test to make sure the two types of errors are triggered https://github.com/swift-server/swift-aws-lambda-runtime/pull/545/commits/156e6c7265346b5e1177293ffaf8b8130685a88d

Reply:
Hello @adam-fowler can you check the updated test? It covers the two behaviours I’m observing (connection lost in `.idle` state and `.waitingForNextInvocation` state.

Reply:
@adam-fowler Good catch.  I moved the promise fulfilment code to `channelInactive` using the code you suggested.  You're correct - this addresses the root cause